### PR TITLE
refactor: del gradlePlugins rocker

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/Gradle.java [hunk 1]
```diff
@@ -59,6 +61,22 @@ public void apply(GeneratorContext generatorContext) {
...
+            }
+        }
+        if (generatorContext.getFeatures().language().isGroovy() || generatorContext.getFeatures().testFramework().isSpock()) {
+            generatorContext.addBuildPlugin(GradlePlugin.builder().id("groovy").build());
+        }
+        generatorContext.addBuildPlugin(GradlePlugin.builder().id(
+                (generatorContext.getFeatures().mainClass().isPresent() ||
+                        generatorContext.getFeatures().contains("oracle-function") ||
+                        generatorContext.getApplicationType() == ApplicationType.DEFAULT && generatorContext.getFeatures().contains("aws-lambda")) ?
+                "io.micronaut.application" : "io.micronaut.library").lookupArtifactId("micronaut-gradle-plugin").build());
```
Comment (ilopmar) [inline starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/Gradle.java]: Can you please use an `if` or extract it to a private method or variable? It is difficult to read with the ternary operator inside the `.id(....)` method in the builder.

File: starter-core/src/main/resources/pom.xml [hunk 2]
```diff
@@ -25,17 +25,17 @@
...
+            <version>1.4.32</version>
         </dependency>
         <dependency>
             <groupId>org.jetbrains.kotlin</groupId>
             <artifactId>kotlin-gradle-plugin</artifactId>
-            <version>1.4.31</version>
+            <version>1.4.32</version>
         </dependency>
         <dependency>
             <groupId>io.micronaut.gradle</groupId>
```
Comment (ilopmar) [inline starter-core/src/main/resources/pom.xml:41]: Latest version of the micronaut-gradle-plugin is 1.4.6.

**Review by ilopmar — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/Gradle.java (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/Gradle.java]: I have extracted a method.

File: starter-core/src/main/resources/pom.xml (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/main/resources/pom.xml:41]: I have set [micronaut gradle plugin](https://plugins.gradle.org/plugin/io.micronaut.library) to 1.4.6

**Review by ilopmar — APPROVED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/lang/kotlin/Kotlin.java [hunk 3]
```diff
@@ -55,7 +55,7 @@ public void processSelectedFeatures(FeatureContext featureContext) {
 
     @Override
     public void apply(GeneratorContext generatorContext) {
-        generatorContext.getBuildProperties().put("kotlinVersion", "1.4.30");
+        generatorContext.getBuildProperties().put("kotlinVersion", "1.4.32");
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/lang/kotlin/Kotlin.java:58]: Why is this hard coded?

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/lang/kotlin/Kotlin.java:58]: It was hardcoded before this PR. I just bump up the version. But I will extract it to the pom.xml and we can get rid of this magic number.

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/lang/kotlin/Kotlin.java:58]: I created a separate PR https://github.com/micronaut-projects/micronaut-starter/pull/747

**Pull request merged**